pycsw/ogc/csw/csw3.py

```diff
@@ -2053,7 +2052,7 @@ def resolve_nsmap(self, list_):
                 prefix = tn.split(':')[0]
                 if prefix in nsmap.keys():  # get uri
                     uri = nsmap[prefix]
-                    newprefix = self.parent.context.namespaces.keys()[self.parent.context.namespaces.values().index(uri)]
+                    newprefix = list(self.parent.context.namespaces.keys())[list(self.parent.context.namespaces.values()).index(uri)]
```

Comment:
Probably keys and values iterate in the same order, but this seems a bit fragile to me. Why not just:
`next(k for k, v in self.parent.context.namespaces.items() if v == uri)`